src/main/java/io/reactivex/internal/operators/parallel/ParallelJoin.java

```diff
@@ -238,9 +238,7 @@ void drainLoop() {
 
                     boolean empty = true;
 
-                    for (int i = 0; i < n; i++) {
-                        JoinInnerSubscriber<T> inner = s[i];
-
+                    for (JoinInnerSubscriber<T> inner : s) {
```

Comment:
Hmm, yeah, I forget that who knows what optimization the Android VM does (the JVM's escape analysis optimizes this nicely). I'll restore the original indexed version.